packages/attestation-service/src/requestHandlers/attestation.ts

```diff
@@ -111,6 +117,24 @@ class AttestationRequestHandler {
     // TODO: Check expiration
   }
 
+  private async verifyPepperIfApplicable(): Promise<void> {
+    if (this.attestationRequest.blindedPhoneNumberSignature) {
+      try {
+        const phoneNumberHashDetails = await getPhoneNumberIdentifierFromSignature(
+          this.attestationRequest.phoneNumber,
+          this.attestationRequest.blindedPhoneNumberSignature,
+          blsBlindingClient
+        )
+        if (phoneNumberHashDetails.pepper !== this.attestationRequest.salt) {
```

Comment:
It might be worth adding a comment here stating that the (salt+phone_number) are verified to match the on-chain identifier elsewhere in the attestation service.